### PR TITLE
Support remote image link

### PR DESCRIPTION
**Pull request by dieterplex**
Based on the work of https://github.com/Michael-F-Bryan/mdbook-epub/pull/25, but choosing lightweight approach.
And found some metrics interesting from https://github.com/Schniz/fnm/pull/547.

cooking tests

### CODE REVIEW

**Comment by dieterplex** [comment]
Tested with these books having remote linked images
- https://github.com/rust-lang/edition-guide
- https://github.com/rustwasm/book

**Comment by blandger** [comment]
Is it finished, correct?

**Comment by dieterplex** [comment]
Yes, please have a review.

**Comment by blandger** [comment]
I will look soon.

File: src/resources.rs [hunk 1]
```diff
@@ -107,34 +161,104 @@ fn assets_in_markdown(src: &str, parent_dir: &Path) -> Result<Vec<PathBuf>, Erro
...
+    #[cfg_attr(test, automock)]
+    pub(crate) trait ContentRetriever {
+        fn download(&self, assets: &Asset) -> Result<(), Error>;
+        fn read(&self, path: &Path, buffer: &mut Vec<u8>) -> Result<(), Error>;
+    }
+
+    pub(crate) struct ResourceHandler;
+
+    impl ContentRetriever for ResourceHandler {
+        fn download(&self, asset: &Asset) -> Result<(), Error> {
```
Comment (blandger) [inline src/resources.rs:235]: As I can see `download(..)` is not covered by tests. Could you pls add error (missing remote resource) and happy paths for that ? Mocking remote will be fine I think.

File: src/resources.rs [hunk 2]
```diff
@@ -41,33 +42,88 @@ pub(crate) fn find(ctx: &RenderContext) -> Result<Vec<Asset>, Error> {
...
+        let filename = absolute_location.strip_prefix(dest_dir).unwrap();
+        let asset = Asset::new(filename, &absolute_location, AssetKind::Remote(url));
+        trace!("{:#?}", asset);
+        Ok(asset)
+    }
+
+    fn from_local(link: &str, src_dir: &Path, chapter_path: &Path) -> Result<Asset, Error> {
+        let full_path = src_dir.join(chapter_path);
+        let asset_path = full_path.parent().ok_or_else(|| {
+            Error::AssetFileNotFound(format!(
```
[the remark this answers is not in the document]
Comment (dieterplex) [reply src/resources.rs]: I think it won't be needed and leave comments there.

File: src/resources.rs (hunk 1 above)
Comment (dieterplex) [reply src/resources.rs:227]: Using `AssetFileNotFound`  should be enough I think.

**Comment by dieterplex** [comment]
Separate lints & fmt fix in another PR.

**Pull request merged**